man/C/msec.8
.ds q \N'34'
.TH msec 0.60.5 msec "Mandriva Linux"
.SH NAME
msec \- Mandriva Linux security tools
.SH SYNOPSIS
.nf
.B msec [options]
.B msecperms [options]
.B msecgui [options]
.fi
.SH DESCRIPTION
.B msec
is responsible to maintain system security in Mandriva. It supports different security
configurations, which can be organized into several security levels, stored in
/etc/security/msec/level.LEVELNAME. Currently, three preconfigured security levels are
provided with Mandriva Linux:

.TP
\fBnone\fR
this level disables all msec options. It should be used when you want to manage
all aspects of system security on your own.

.TP
\fBdefault\fR
this is the default security level, which configures a reasonably safe set of security
features. It activates several periodic system checks, and sends the results of their
execution by email (by default, the local 'root' account is used).

.TP
\fBsecure\fR
this level is configured to provide maximum system security, even at the cost of limiting
the remote access to the system, and local user permissions. It also runs a wider set of
periodic checks, enforces the local password settings, and periodically checks if the
system security settings, configured by msec, were modified directly or by some other
application.

.TP
Note that besides those levels you may create as many levels as necessary.

.PP

The security settings are stored in \fB/etc/security/msec/security.conf\fR
file, and default settings for each predefined level are stored in
\fB/etc/security/msec/level.LEVEL\fR.  Permissions for files and directories
that should be enforced or checked for changes are stored in
\fB/etc/security/msec/perms.conf\fR, and default permissions for each
predefined level are stored in \fB/etc/security/msec/perm.LEVEL\fR.  Note
that user-modified parameters take precedence over default level settings. For
example, when default level configuration forbids direct root logins, this
setting can be overridden by the user.

.PP

The following options are supported by msec applications:

.TP
\fBmsec\fR:
.PP

This is the console version of msec. It is responsible for system security configuration
and checking and transitions between security levels.

When executed without parameters, msec will read the system configuration file
(/etc/security/msec/security.conf), and enforce the specified security
settings. The operations are logged to \fB/var/log/msec.log\fP file, and also
to syslog, using \fBLOG_AUTHPRIV\fR facility.  Please note that msec should
by run as root.

\fB\-h, --help\fR
    This option will display the list of supported command line options.

\fB\-l, --level <level>\fR
    List the default configuration for given security level.

\fB\-f, --force <level>\fR

    Apply the specified security level to the system, overwritting all local
changes in /etc/security/msec/security.conf. This usually should be performed
either on first install, on when a transition to a different level is required.

\fB\-d\fR
    Enable debugging messages.

\fB\-p, --pretend\fR
    Verify the actions that will be performed by msec, without actually
doing anything to the system. In this mode of operation, msec performs all the
required tasks, except effectively writting data back to disk.

.TP
\fBmsecperms\fR:
.PP

This application is responsible for system permission checking and enforcements.

When executed without parameters, msecperms will read the permissions
configuration file (/etc/security/msec/perms.conf), and enforce the specified
security settings. The operations are logged to \fB/var/log/msec.log\fP file,
and also to syslog, using \fBLOG_AUTHPRIV\fR facility.  Please note that msecperms
should by run as root.

\fB\-h, --help\fR
    This option will display the list of supported command line options.

\fB\-l, --level <level>\fR
    List the default configuration for given security level.

\fB\-f, --force <level>\fR
    Apply the specified security level to the system, overwritting all local
changes in /etc/security/msec/perms.conf. This usually should be performed
either on first install, on when a transition to a different level is required.

\fB\-e, --enforce\fR
    Enforce the default permissions on all files.

\fB\-d\fR
    Enable debugging messages.

\fB\-p, --pretend\fR
    Verify the actions that will be performed by msec, without actually
doing anything to the system. In this mode of operation, msec performs all the
required tasks, except effectively writting data back to disk.

.TP
\fBmsecgui\fR:
.PP

This is the GTK version of msec. It acts as frontend to all msec functionalities.

\fB\-h, --help\fR
    This option will display the list of supported command line options.

\fB\-d\fR
    Enable debugging messages.

.SH EXAMPLES

\fBEnforce system configuration according to /etc/security/msec/security.conf file:\fP
    msec

\fBDisplay system configuration changes without enforcing anything:\fP
    msec -p

\fBInstall predefined security level 'default':\fP
    msec -f default

\fBPreview changes inflicted by change to 'default' level:\fP
    msec -p -f default

\fBCreate a custom security level based on 'default':\fP
    cp /etc/security/msec/level.default /etc/security/msec/level.my
    edit /etc/security/msec/level.my
    msec -f my

\fBEnforce system permissions according to /etc/security/msec/perms.conf file:\fP
    msecperms

\fBDisplay permissions changes without enforcing anything:\fP
    msecperms -p

\fBInstall predefined permissions for level 'default':\fP
    msecperms -f default

\fBPreview changes inflicted by change to 'default' level:\fP
    msecperms -p -f default

\fBCreate a custom permissions level based on 'secure':\fP
    cp /etc/security/msec/perm.secure /etc/security/msec/perm.my
    edit /etc/security/msec/level.my
    msecperms -f my

.SH "SECURITY OPTIONS"

The following security options are supported by msec:



.TP 4
.B \fIenable_dns_spoofing_protection\fP
Enable/Disable name resolution spoofing protection.

MSEC parameter: \fIENABLE_IP_SPOOFING_PROTECTION\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fImail_empty_content\fP
Enables sending of empty mail reports.

MSEC parameter: \fIMAIL_EMPTY_CONTENT\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIaccept_broadcasted_icmp_echo\fP
Accept/Refuse broadcasted icmp echo.

MSEC parameter: \fIACCEPT_BROADCASTED_ICMP_ECHO\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_sudo\fP
Enables support for sudo application, which allows users to run applications using system account. If yes, users must autenticate themselves using password. If this parameter is set to 'wheel', users must belong to the 'wheel' group to be able to use sudo

MSEC parameter: \fIENABLE_SUDO\fP

Accepted values: \fIyes, no, wheel\fP


.TP 4
.B \fIallow_xserver_to_listen\fP
The argument specifies if clients are authorized to connect to the X server on the tcp port 6000 or not.

MSEC parameter: \fIALLOW_XSERVER_TO_LISTEN\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_chkrootkit\fP
Enables checking for known rootkits using chkrootkit.

MSEC parameter: \fICHECK_CHKROOTKIT\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_suid_root\fP
Enables checking for additions/removals of suid root files.

MSEC parameter: \fICHECK_SUID_ROOT\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_at_crontab\fP
Enable/Disable crontab and at for users. Put allowed users in /etc/cron.allow and /etc/at.allow (see man at(1) and crontab(1)).

MSEC parameter: \fIENABLE_AT_CRONTAB\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIaccept_bogus_error_responses\fP
Accept/Refuse bogus IPv4 error messages.

MSEC parameter: \fIACCEPT_BOGUS_ERROR_RESPONSES\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_suid_md5\fP
Enables checksum verification for suid files.

MSEC parameter: \fICHECK_SUID_MD5\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fImail_user\fP
Defines email to receive security notifications.

MSEC parameter: \fIMAIL_USER\fP

Accepted values: \fI*\fP


.TP 4
.B \fIallow_autologin\fP
Allow/Forbid autologin.

MSEC parameter: \fIALLOW_AUTOLOGIN\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_pam_wheel_for_su\fP
Enabling su only from members of the wheel group or allow su from any user.

MSEC parameter: \fIENABLE_PAM_WHEEL_FOR_SU\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcreate_server_link\fP
Creates the symlink /etc/security/msec/server to point to /etc/security/msec/server.SERVER_LEVEL. The /etc/security/msec/server is used by chkconfig --add to decide to add a service if it is present in the file during the installation of packages.

MSEC parameter: \fICREATE_SERVER_LINK\fP

Accepted values: \fIno, default, secure\fP


.TP 4
.B \fIset_shell_timeout\fP
Set the shell timeout. A value of zero means no timeout.

MSEC parameter: \fISHELL_TIMEOUT\fP

Accepted values: \fI*\fP


.TP 4
.B \fIcheck_user_files\fP
Enables permission checking on users' files that should not be owned by someone else, or writable.

MSEC parameter: \fICHECK_USER_FILES\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_shadow\fP
Enables checking for empty passwords.

MSEC parameter: \fICHECK_SHADOW\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_password\fP
Use password to authenticate users. Take EXTREMELY care when disabling passwords, as it will leave the machine COMPLETELY vulnerable.

MSEC parameter: \fIENABLE_PASSWORD\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIset_win_parts_umask\fP
Set umask option for mounting vfat and ntfs partitions. A value of None means default umask.

MSEC parameter: \fIWIN_PARTS_UMASK\fP

Accepted values: \fIno, *\fP


.TP 4
.B \fIcheck_open_port\fP
Enables checking for open network ports.

MSEC parameter: \fICHECK_OPEN_PORT\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_log_strange_packets\fP
Enable/Disable the logging of IPv4 strange packets.

MSEC parameter: \fIENABLE_LOG_STRANGE_PACKETS\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_rpm\fP
Enables verification of installed packages.

MSEC parameter: \fICHECK_RPM\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_pam_root_from_wheel\fP
Allow root access without password for the members of the wheel group.

MSEC parameter: \fIENABLE_PAM_ROOT_FROM_WHEEL\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fImail_warn\fP
Enables security results submission by email.

MSEC parameter: \fIMAIL_WARN\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIpassword_length\fP
Set the password minimum length and minimum number of digit and minimum number of capitalized letters.

MSEC parameter: \fIPASSWORD_LENGTH\fP

Accepted values: \fI*\fP


.TP 4
.B \fIset_root_umask\fP
Set the root umask.

MSEC parameter: \fIROOT_UMASK\fP

Accepted values: \fI*\fP


.TP 4
.B \fIcheck_sgid\fP
Enables checking for additions/removals of sgid files.

MSEC parameter: \fICHECK_SGID\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_promisc\fP
Activate/Disable ethernet cards promiscuity check.

MSEC parameter: \fICHECK_PROMISC\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_apparmor\fP
Enables support for AppArmor security framework

MSEC parameter: \fIENABLE_APPARMOR\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIallow_x_connections\fP
Allow/Forbid X connections. Accepted arguments: yes (all connections are allowed), local (only local connection), no (no connection).

MSEC parameter: \fIALLOW_X_CONNECTIONS\fP

Accepted values: \fIyes, no, local\fP


.TP 4
.B \fIcheck_writable\fP
Enables checking for files/directories writable by everybody.

MSEC parameter: \fICHECK_WRITABLE\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_console_log\fP
Enable/Disable syslog reports to console terminal 12.

MSEC parameter: \fIENABLE_CONSOLE_LOG\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_ip_spoofing_protection\fP
Enable/Disable IP spoofing protection.

MSEC parameter: \fIENABLE_DNS_SPOOFING_PROTECTION\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIbase_level\fP
Defines the base security level, on top of which the current configuration is based.

MSEC parameter: \fIBASE_LEVEL\fP

Accepted values: \fI*\fP


.TP 4
.B \fIcheck_perms\fP
Enables periodic permission checking for system files.

MSEC parameter: \fICHECK_PERMS\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIset_shell_history_size\fP
Set shell commands history size. A value of -1 means unlimited.

MSEC parameter: \fISHELL_HISTORY_SIZE\fP

Accepted values: \fI*\fP


.TP 4
.B \fIallow_reboot\fP
Allow/Forbid system reboot and shutdown to local users.

MSEC parameter: \fIALLOW_REBOOT\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIsyslog_warn\fP
Enables logging to system log.

MSEC parameter: \fISYSLOG_WARN\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_shosts\fP
Enables checking for dangerous options in users' .rhosts/.shosts files.

MSEC parameter: \fICHECK_SHOSTS\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_passwd\fP
Enables password-related checks, such as empty passwords and strange super-user accounts.

MSEC parameter: \fICHECK_PASSWD\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIpassword_history\fP
Set the password history length to prevent password reuse. This is not supported by pam_tcb.

MSEC parameter: \fIPASSWORD_HISTORY\fP

Accepted values: \fI*\fP


.TP 4
.B \fIcheck_security\fP
Enables daily security checks.

MSEC parameter: \fICHECK_SECURITY\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIallow_root_login\fP
Allow/Forbid direct root login.

MSEC parameter: \fIALLOW_ROOT_LOGIN\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIcheck_unowned\fP
Enables checking for unowned files.

MSEC parameter: \fICHECK_UNOWNED\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIallow_user_list\fP
Allow/Forbid the list of users on the system on display managers (kdm and gdm).

MSEC parameter: \fIALLOW_USER_LIST\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fInotify_warn\fP
Enables support for security notifications using libnotify. This allows the security notifications to be delivered directly to the users' desktop

MSEC parameter: \fINOTIFY_WARN\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIallow_remote_root_login\fP
Allow/Forbid remote root login via sshd. You can specify yes, no and without-password. See sshd_config(5) man page for more information.

MSEC parameter: \fIALLOW_REMOTE_ROOT_LOGIN\fP

Accepted values: \fIyes, no, without_password\fP


.TP 4
.B \fIenable_msec_cron\fP
Enable/Disable msec hourly security check.

MSEC parameter: \fIENABLE_MSEC_CRON\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_sulogin\fP
Enable/Disable sulogin(8) in single user level.

MSEC parameter: \fIENABLE_SULOGIN\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIallow_xauth_from_root\fP
Allow/forbid to export display when passing from the root account to the other users. See pam_xauth(8) for more details.

MSEC parameter: \fIALLOW_XAUTH_FROM_ROOT\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIset_user_umask\fP
Set the user umask.

MSEC parameter: \fIUSER_UMASK\fP

Accepted values: \fI*\fP


.TP 4
.B \fIaccept_icmp_echo\fP
Accept/Refuse icmp echo.

MSEC parameter: \fIACCEPT_ICMP_ECHO\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIauthorize_services\fP
Configure access to tcp_wrappers services (see hosts.deny(5)).  If arg = yes, all services are authorized. If arg = local, only local ones are, and if arg = no, no services are authorized. In this case, To authorize the services you need, use /etc/hosts.allow (see hosts.allow(5)).

MSEC parameter: \fIAUTHORIZE_SERVICES\fP

Accepted values: \fIyes, no, local\fP


.TP 4
.B \fItty_warn\fP
Enables periodic security check results to terminal.

MSEC parameter: \fITTY_WARN\fP

Accepted values: \fIyes, no\fP


.TP 4
.B \fIenable_policykit\fP
Enables support for PolicyKit framework, which allows ordinary users to run system application

MSEC parameter: \fIENABLE_POLICYKIT\fP

Accepted values: \fIyes, no\fP

.RE
.SH NOTES
Msec applications must be run by root.
.SH AUTHORS
Frederic Lepied

Eugeni Dodonov <eugeni@mandriva.com>



NEWS
Version 0.60.8 - January 26 2009, Eugeni Dodonov
- Changed without_password to without-password to prevent bogus errors.
- Running expensive msec_find only when required.
- Fixing permissions on msec-created files (#27820 #47059)
- Handling network settings as in previous msec versions (#47240).
- Added default response to msecgui Save dialog.
- Implemented support for custom paths checks in msecperms.

Version 0.60.7 - January 21 2009, Eugeni Dodonov
- Now correctly integrating with MCC.

Version 0.60.6 - January 20 2009, Eugeni Dodonov
- Removed Authentication tab (now handled by a separate application)
- Now it is possible to save settings without quitting.
- Better detection for file modifications (such as symlinks, moves, etc)
- Now asking to save changes before quitting when necessary.
- Highlighting default option value according to current level.
- Level selection improvements.
- Checking for $DISPLAY variable.
- Added HAL to list of save services.
- Now highlighting options which are different from default values for level.
- Improved GUI spacing between options.
- Removed Notifications tab (merged with initial screen and periodic
  checks screen).
- Better handling of non-existent files (inittab and sysctl).

Version 0.60.5 - January 14 2009, Eugeni Dodonov
- Fixed msecperms -e (setting default permissions to files).

Version 0.60.4 - January 13 2009, Eugeni Dodonov
- Updated gui to allow immediate preview of options on level change.
- New permissions control GUI.
- Added support for custom security levels.

Version 0.60.3 - January 07 2009, Eugeni Dodonov
- Bugfixes for gdm config handling.
- Implemented authentication gui.
- Added support for --embedded.
- Now using /etc/security/shell instead of /etc/sysconfig/msec.

Version 0.60.2 - January 07 2009, Eugeni Dodonov
- Bugfixes for kdmrc handling.

Version 0.60.1 - January 07 2009, Eugeni Dodonov

- Complete msec redesign for Mandriva 2009.1.

Version 0.50.11 - 16 December 2008, Eugeni Dodonov

- Correctly handle permit_root_login in sshd_config on level change
  (#19726).
- Handle multibyte characters in msec reports (#26773).

Version 0.50.10 - 01 October 2008, Thierry Vignaud

- cron entry:
  o blacklist cifs instead of only smbfs for samba
  o exclude /media from searching like /mnt is
  o run with idle IOnice priority (#42795)


src/msec/msec.py
#!/usr/bin/python -O
"""This is the main msec module.
It checks/sets the security levels, configures security variables,
and works as a frontend to libmsec.
"""

import sys
import os
import string
import getopt
import gettext
import imp
import re

# config
import config

# version
try:
    from version import version
except:
    version = "development version"

# libmsec
from libmsec import MSEC, Log

import logging

# localization
try:
    cat = gettext.Catalog('msec')
    _ = cat.gettext
except IOError:
    _ = str

# {{{ usage
def usage():
    """Prints help message"""
    print """Msec: Mandriva Security Center (%s).

When run without parameters, msec will read the configuration from
/etc/security/msec/msec.conf, and enforce the specified security settings.
If no configuration file is found on the system, a default configuration
will be created.

Arguments to msec:
    -h, --help              displays this helpful message.
    -l, --level <level>     displays configuration for specified security
                            level.
    -f, --force <level>     force new level, overwriting user settings.
    -d                      enable debugging messages.
    -p, --pretend           only pretend to change the level, perform no real
                            actions. Use this to see what operations msec
                            will perform.
""" % version
# }}}

if __name__ == "__main__":
    # default options
    force_level = False
    log_level = logging.INFO
    commit = True

    # parse command line
    try:
        opt, args = getopt.getopt(sys.argv[1:], 'hl:f:dp', ['help', 'list=', 'force=', 'debug', 'pretend'])
    except getopt.error:
        usage()
        sys.exit(1)
    for o in opt:
        # help
        if o[0] == '-h' or o[0] == '--help':
            usage()
            sys.exit(0)
        # list
        elif o[0] == '-l' or o[0] == '--list':
            level = o[1]
            log = Log(interactive=True, log_syslog=False, log_file=False)
            levelconf = config.load_defaults(log, level)
            params = levelconf.list_options()
            if not params:
                print >>sys.stderr, _("Invalid security level '%s'.") % level
                sys.exit(1)
            for item in params:
                print "%s=%s" % (item, levelconf.get(item) )
            sys.exit(0)
        # force new level
        elif o[0] == '-f' or o[0] == '--force':
            level = o[1]
            force_level = True
        # debugging
        elif o[0] == '-d' or o[0] == '--debug':
            log_level = logging.DEBUG
        # check-only mode
        elif o[0] == '-p' or o[0] == '--pretend':
            commit = False

    # verifying use id
    if os.geteuid() != 0:
        print >>sys.stderr, _("Msec: Mandriva Security Center (%s)\n") % version
        print >>sys.stderr, _("Error: This application must be executed by root!")
        print >>sys.stderr, _("Run with --help to get help.")
        sys.exit(1)

    # configuring logging
    interactive = sys.stdin.isatty()
    if interactive:
        # logs to file and to terminal
        log = Log(log_path=config.SECURITYLOG, interactive=True, log_syslog=False, log_level=log_level)
    else:
        # TODO: review logging messages
        #log_level = logging.WARN
        log = Log(log_path=config.SECURITYLOG, interactive=False, log_syslog=False, log_level=log_level)

    # loading initial config
    msec_config = config.MsecConfig(log, config=config.SECURITYCONF)

    # forcing new level
    if force_level:
        # first load the default configuration for level
        levelconf = config.load_defaults(log, level)
        params = levelconf.list_options()
        if not params:
            log.error(_("Level '%s' not found, aborting.") % level)
            sys.exit(1)
        log.info(_("Switching to '%s' level.") % level)
        for opt in params:
            msec_config.set(opt, levelconf.get(opt))
    else:
        msec_config.load()

    # load the msec library
    msec = MSEC(log)

    # apply the config to msec
    msec.apply(msec_config)
    # writing back changes
    msec.commit(commit)
    # saving updated config
    if force_level and commit:
        if not msec_config.save():
            log.error(_("Unable to save config!"))
    sys.exit(0)
